### PR TITLE
added content type documentation and fixed a few bugs

### PR DESCRIPTION
**Pull request by danrot**
This PR includes a basic documentation for all content types.
Additionally it restructures the documentation as discussed with @chirimoya on IRC.

@dantleech Can you please review?

### CODE REVIEW

File: reference/content-types/category_list.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+Parameters
+----------
+
+No parameters available
+
+Example
+-------
+
+.. code-block:: xml
```
Comment (dantleech) [inline reference/content-types/category_list.rst]: I think it would be beneficial to show the structure of the output on the frontend. wdyt?

File: reference/content-types/index.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                    <title lang="en">Resourcelocator</title>
+                </meta>
+
+                <tag name="sulu.rlp"/>
+            </property>
+        </properties>
+    </template>
+
+This chapter will describe which types you can insert within the
+``properties``-tag. Every content type in the documentation comes with an
```
Comment (dantleech) [inline reference/content-types/index.rst]: Remove the `-` should read

````
``properties`` tag
````

(not even sure if it will render when you hyphenate an inline code thing)

File: reference/content-types/index.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+example ``property``-tag to clarify the usage.
```
Comment (dantleech) [inline reference/content-types/index.rst]: Remove `-`

File: reference/content-types/index.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    phone
+    date
+    time
+    password
+    color
+    checkbox
+    snippet
+    category_list
+    tag_list
+    location
```
Comment (dantleech) [inline reference/content-types/index.rst]: Should be sorted alphabetically. in vim select visually and type `:sort`

File: reference/content-types/internal_links.rst [hunk 5]
```diff
@@ -0,0 +1,41 @@
+Internal links
+==============
+
+Description
+-----------
+
+Shows a list with the possibility to add links to other sites managed in Sulu.
```
Comment (dantleech) [inline reference/content-types/internal_links.rst]: other sites or other pages?

File: reference/content-types/internal_links.rst [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+.. list-table::
+    :header-rows: 1
+
+    * - Parameter
+      - Type
+      - Description
+    * - properties
+      - collection
+      - Defines with which key which property of the linked page should be
+        populated to the HTML template.
```
Comment (dantleech) [inline reference/content-types/internal_links.rst]: Thats a nice way of creating a table.

File: reference/content-types/location.rst [hunk 7]
```diff
@@ -0,0 +1,42 @@
+Location
+========
+
+Description
+-----------
+
+Adds the possibility to assign geographic information to a page. Can be used in
+combination with Google Maps and Open Street Maps.
```
Comment (dantleech) [inline reference/content-types/location.rst]: ``Can be used either with..``

File: reference/content-types/smart_content.rst [hunk 8]
```diff
@@ -0,0 +1,71 @@
+Smart content
+=============
+
+Description
+-----------
+
+Shows a list of pages, which depends on a configureable filter. You can define
```
Comment (dantleech) [inline reference/content-types/smart_content.rst]: `which depend on a ..` 
`s/configureable/configurable`

File: reference/content-types/smart_content.rst [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+which which site's children should be included, what tags the filtered pages
```
Comment (dantleech) [inline reference/content-types/smart_content.rst]: `You can include branches of the content sites tree and tagged content, the results can be sorted and limited` .. or something. At least it should not say `which which` :)

File: reference/content-types/smart_content.rst [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+.. list-table::
+    :header-rows: 1
+
+    * - Parameter
+      - Type
+      - Description
+    * - max_per_page
+      - integer
+      - Limits the results per page. This parameter also enables the paging of
+        the results at all.
```
Comment (dantleech) [inline reference/content-types/smart_content.rst]: `Limits the results per page. Set to `---` (?) to disable pagination.`

File: reference/content-types/smart_content.rst [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+    * - page_parameter
+      - string
+      - Defines the page parameter for the query string on the website.
```
Comment (dantleech) [inline reference/content-types/smart_content.rst]: `Defines the page number key to be used in the query string` ?

File: reference/content-types/smart_content.rst [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    * - properties
+      - collection
+      - Defines with which key which property of the filtered page should be
```
Comment (dantleech) [inline reference/content-types/smart_content.rst]: Not sure what this means `Defines the property names which will be exposed in the website template` ?

File: reference/content-types/tag_list.rst [hunk 13]
```diff
@@ -0,0 +1,26 @@
+Tag list
+========
+
+Description
+-----------
+
+Shows a simple text line with an autocomplete feature for the available Tags in
+the system. Tags, which do not already exist, will be created. Tags can be
```
Comment (dantleech) [inline reference/content-types/tag_list.rst]: No `,`. `Tags which do not already exist will be created.`.

You could also do this as a note

````
.. note::

    Tags which do not already exist will be created.
````

File: reference/content-types/text_editor.rst [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+
+.. list-table::
+    :header-rows: 1
+
+    * - Parameter
+      - Type
+      - Description
+    * - godMode
+      - boolean
+      -
```
Comment (dantleech) [inline reference/content-types/text_editor.rst]: Nobody knows what this does I guess

Comment (dantleech) [inline reference/content-types/text_editor.rst]: I would leave it out

**Comment by dantleech** [comment]
Awesome :) It would be great to show how the content is formatted in the HTML template, but we could do that later.

File: reference/content-types/category_list.rst (hunk 1 above)
Comment (danrot) [inline reference/content-types/category_list.rst]: Definitely, but that would be another PR for me. This documentation is already better than nothing :-)

File: reference/content-types/index.rst (hunk 4 above)
Comment (danrot) [inline reference/content-types/index.rst]: Hm, actually I tried to group them a bit. But alphabetically is also ok for me, will change it.

File: reference/content-types/internal_links.rst (hunk 5 above)
Comment (danrot) [inline reference/content-types/internal_links.rst]: Pages, will change it

File: reference/content-types/internal_links.rst (hunk 6 above)
Comment (danrot) [inline reference/content-types/internal_links.rst]: Thank you ;-)

File: reference/content-types/smart_content.rst (hunk 10 above)
Comment (danrot) [inline reference/content-types/smart_content.rst]: I'll change it to `Don't set this parameter to disable pagination`

File: reference/content-types/smart_content.rst (hunk 12 above)
Comment (danrot) [inline reference/content-types/smart_content.rst]: Yeah, this is better, I'll change it

File: reference/content-types/text_editor.rst (hunk 14 above)
Comment (danrot) [inline reference/content-types/text_editor.rst]: I guess @marcelmoosbrugger should know ;-) I can omit it for now.

File: reference/content-types/smart_content.rst (hunk 10 above)
Comment (dantleech) [inline reference/content-types/smart_content.rst]: `If this parameter is not specified, pagination will be disabled` or `Omit this parameter to disable pagination`

**Comment by danrot** [comment]
@dantleech Fixed your issues, as already said, I would add the format of the content later, as it is already better than the current situation :-)
